what4/src/What4/Protocol/Online.hs

```diff
@@ -440,3 +483,39 @@ getSatResult yp = do
                   , "*** standard error:"
                   , LazyText.unpack txt
                   ]
+
+
+-- | If the solver cannot voluntarily limit itself to the requested
+-- timeout period, this runs a local async process with a slightly
+-- longer time period that will forcibly terminate the solver process
+-- if it expires while the solver process is still running.
+--
+-- Note that this will require re-establishment of the solver process
+-- and any associated context for any subsequent solver goal
+-- evaluation.
+
+withLocalGoalTimeout ::
+  SolverProcess t s
+  -> (WriterConn t s -> IO (SatResult () ()))
+  -> IO (Either SomeException (SatResult () ()))
+withLocalGoalTimeout solverProc action =
+  if getGoalTimeoutInSeconds (solverGoalTimeout solverProc) == 0
```

Comment:
The comment above says that `solverGoalTimeout` is only advisory?

Reply:
Fixed the comment as described in the previous note.